apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex

```diff
@@ -182,30 +188,28 @@ defmodule AndiWeb.EditLiveView.DataDictionaryForm do
 
     case Jason.decode(file) do
       {:error, _} ->
-        new_changeset =
-          changeset
-          |> Map.put(:action, :update)
-          |> Changeset.add_error(:schema_sample, "There was a problem interpreting this file")
-
-        {:noreply, assign(socket, changeset: new_changeset, loading_schema: false)}
+        changeset
+        |> send_error_interpreting_file(socket)
 
       {:ok, decoded_json} ->
-        new_changeset =
-          decoded_json
-          |> List.wrap()
-          |> DataDictionaryFormSchema.changeset_from_file(socket.assigns.dataset_id)
-
-        assign_new_schema(socket, new_changeset)
+        try do
+          new_changeset =
+            decoded_json
+            |> Enum.empty?()
+            |> if(do: throw(:error), else: decoded_json)
+            |> List.wrap()
+            |> DataDictionaryFormSchema.changeset_from_file(socket.assigns.dataset_id)
```

Comment:
This is an unconventional pattern. Feels weird to swap the original value of the pipeline back into it further down.